---
title: "<img src='./inst/hex/orthogene.png' height='400'><br>" 
author: "<h4>Author: <i>Brian M. Schilder</i></h4>" 
date: "<h4>Most recent update: <i>`r format( Sys.Date(), '%b-%d-%Y')`</i></h4>"
output:
  github_document 
editor_options: 
  chunk_output_type: inline
---

## Interspecies gene mapping 

`orthogene` is an R package for easy mapping of orthologous genes across hundreds of species.   
It pulls up-to-date interspecies gene ortholog mappings across 700+ organisms. 

It also provides various utility functions to map common objects 
(e.g. data.frames, gene expression matrices, lists) 
onto 1:1 gene orthologs from any other species. 


## [Documentation website](https://neurogenomics.github.io/orthogene/) 

# Installation

```{r, eval=FALSE}
if(!"remotes" %in% rownames(installed.packages())) install.packages("remotes")

remotes::install_github("neurogenomics/orthogene")
```

```{r setup}
library(orthogene)

method <- "homologene"
```

# Methods 

For most functions, `orthogene` lets users choose between two different methods, each with complementary strengths and weaknesses: `"gprofiler"` and `"homologene"` 

In general, we recommend you use `"gprofiler"` when possible, as it tends to be more comprehensive. 

It's also worth noting that for smaller gene sets, the speed difference between these methods becomes negligible. 

```{r, echo=FALSE}
pros_cons <- data.frame(gprofiler=c("Reference organisms"="700+",
                       "Gene mappings"="More comprehensive",
                       "Updates"="Frequent", 
                       "Orthology databases"=paste("Ensembl",
                                                   "HomoloGene",
                                                   "WormBase",sep = ", "),
                       "Data location"="Remote",
                       "Internet connection"="Required",
                       "Speed"="Slower"),
                       
           homologene=c("# reference organisms"="20+", 
                        "Gene mappings"="Less comprehensive",
                        "Updates"="Less frequent",
                        "Orthology databases"="HomoloGene",
                        "Data location"="Local",
                        "Internet connection"="Not required",
                        "Speed"="Faster")
           )
knitr::kable(pros_cons)
```



# Quick example

## Convert orthologs 

`convert_orthologs` is very flexible with its `gene_df` input,
and can take a data.frame/data.table/tibble, (sparse) matrix, or list/vector containing genes.

Genes will be recognised in most formats (e.g. HGNC, Ensembl, UCSC) 
and can even be a mixture of different formats. 

All genes will be mapped to gene symbols, unless specified otherwise with the
`...` arguments (see `?orthogene::convert_orthologs` for details).  


```{r}
data("exp_mouse")
gene_df <- convert_orthologs(gene_df = exp_mouse,
                             gene_input = "rownames", 
                             gene_output = "rownames", 
                             input_species = "mouse",
                             method = method) 

knitr::kable(as.matrix(head(gene_df)))
```

## Map species

`map_species` lets you standardise species names from a wide variety of identifiers 
(e.g. common name, taxonomy ID, Latin name, partial match). 

All exposed `orthogene` functions (including `convert_orthologs`) 
use `map_species` under the hood, so you don't have to worry about 
getting species names exactly right.   

You can check the full list of available species by simply running 
`map_species()` with no arguments, 
or checking [here](https://biit.cs.ut.ee/gprofiler/page/organism-list).

```{r}
species <- map_species(species = c("human",9544,"mus musculus","fruit fly","Celegans"), 
                       output_format = "scientific_name")
print(species)
```

## Report orthologs 

It may be helpful to know the maximum expected number of orthologous 
gene mappings from one species to another.

`ortholog_report` generates a report that tells you this information 
genome-wide. 

```{r} 
orth.zeb <- report_orthologs(target_species = "zebrafish",
                             reference_species = "human",
                             method_all_genes = method,
                             method_convert_orthologs = method) 
```


## Get all genes 

You can also quickly get all known genes from the genome of a given species.

```{r}
genome_mouse <- all_genes(species = "mouse", 
                          method = method)

knitr::kable(head(genome_mouse))
```


# Session Info 

<details> 

```{r Session Info}
utils::sessionInfo()
```

</details>  


# Related projects

## Tools 

- [`homologene`](https://github.com/oganm/homologene): `orthogene` uses this package. Provides API access to NCBI [HomoloGene](https://www.ncbi.nlm.nih.gov/homologene) database. 

- [`gprofiler2`](https://cran.r-project.org/web/packages/gprofiler2/vignettes/gprofiler2.html): `orthogene` uses this package. `gprofiler2::gorth()` pulls from [many orthology mapping databases](https://biit.cs.ut.ee/gprofiler/page/organism-list).  

- [`annotationTools`](https://www.bioconductor.org/packages/release/bioc/html/annotationTools.html): For interspecies microarray data.  

- [`orthology`](https://www.leibniz-hki.de/en/orthology-r-package.html): R package for ortholog mapping (deprecated?). 

- [`hpgltools::load_biomart_orthologs()`](https://rdrr.io/github/elsayed-lab/hpgltools/man/load_biomart_orthologs.html): Helper function to get orthologs from biomart. 

- [`JustOrthologs`](https://github.com/ridgelab/JustOrthologs/): Ortholog inference from multi-species genomic sequences. 

- [`orthologr`](https://github.com/drostlab/orthologr): Ortholog inference from multi-species genomic sequences.  

- [`OrthoFinder`](https://github.com/davidemms/OrthoFinder): Gene duplication event inference from multi-species genomics.  


## Databases  

- [HomoloGene](https://www.ncbi.nlm.nih.gov/homologene): NCBI database that the R package [homologene](https://github.com/oganm/homologene) pulls from.   

- [gProfiler](https://biit.cs.ut.ee/gprofiler): Web server for functional enrichment analysis and conversions of gene lists.  

- [OrtholoGene](http://orthologene.org/resources.html): Compiled list of gene orthology resources. 




