<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Interspecies gene mapping  • orthogene</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Interspecies gene mapping ">
<meta property="og:description" content="
    orthogene is an R package for easy mapping of orthologous genes across hundreds of species. 
    It pulls up-to-date interspecies gene ortholog mappings across 700+ organisms.
    It also provides various utility functions to map common objects 
    (e.g. data.frames, gene expression matrices, lists) 
    onto 1:1 gene orthologs from any other species. ">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">orthogene</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/benchmarks.html">benchmarks</a>
    </li>
    <li>
      <a href="articles/hexSticker.html">hexSticker</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/neurogenomics/orthogene/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div id="-" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#-" class="anchor"></a><img src="./inst/hex/orthogene.png" height="400"><br>
</h1></div>
<h4>
Author: <i>Brian M. Schilder</i>
</h4>
<h4>
Most recent update: <i>Jul-31-2021</i>
</h4>
<!-- badges: start -->
<!-- badger::badge_codecov() -->
<!-- copied from MungeSumstats README.Rmd -->
<!-- badger::badge_lifecycle("stable", "green") -->
<!-- badger::badge_last_commit()  -->
<!-- badger::badge_license() -->

<div id="interspecies-gene-mapping" class="section level2">
<h2 class="hasAnchor">
<a href="#interspecies-gene-mapping" class="anchor"></a>Interspecies gene mapping</h2>
<p><code>orthogene</code> is an R package for easy mapping of orthologous genes across hundreds of species.<br>
It pulls up-to-date interspecies gene ortholog mappings across 700+ organisms.</p>
<p>It also provides various utility functions to map common objects (e.g. data.frames, gene expression matrices, lists) onto 1:1 gene orthologs from any other species.</p>
<p>In brief, <code>orthogene</code> lets you easily:</p>
<ul>
<li><a href="https://github.com/neurogenomics/orthogene#convert-orthologs"><strong><code>convert_orthologs</code></strong> between any two species</a></li>
<li>
<a href="https://github.com/neurogenomics/orthogene#map-species"><strong><code>map_species</code></strong> names onto standard taxonomic ontologies</a><br>
</li>
<li><a href="https://github.com/neurogenomics/orthogene#report-orthologs"><strong><code>report_orthologs</code></strong> between any two species</a></li>
<li><a href="https://github.com/neurogenomics/orthogene#map-genes"><strong><code>map_genes</code></strong> onto standard ontologies</a></li>
<li><a href="https://github.com/neurogenomics/orthogene#get-all-genes">get <strong><code>all_genes</code></strong> from any species</a></li>
</ul>
</div>
<div id="documentation-website" class="section level2">
<h2 class="hasAnchor">
<a href="#documentation-website" class="anchor"></a><a href="https://neurogenomics.github.io/orthogene/">Documentation website</a>
</h2>
</div>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"remotes"</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span>

<span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"neurogenomics/orthogene"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/neurogenomics/orthogene">orthogene</a></span><span class="op">)</span>

<span class="co"># Setting to "homologene" for the purposes of quick demonstration.</span>
<span class="co"># We generally recommend using method="gprofiler" (default).</span>
<span class="va">method</span> <span class="op">&lt;-</span> <span class="st">"homologene"</span></code></pre></div>
</div>
<div id="methods" class="section level1">
<h1 class="hasAnchor">
<a href="#methods" class="anchor"></a>Methods</h1>
<p>For most functions, <code>orthogene</code> lets users choose between two different methods, each with complementary strengths and weaknesses: <code>"gprofiler"</code> and <code>"homologene"</code></p>
<p>In general, we recommend you use <code>"gprofiler"</code> when possible, as it tends to be more comprehensive.</p>
<p>It’s also worth noting that for smaller gene sets, the speed difference between these methods becomes negligible.</p>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="left">gprofiler</th>
<th align="left">homologene</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Reference organisms</td>
<td align="left">700+</td>
<td align="left">20+</td>
</tr>
<tr class="even">
<td align="left">Gene mappings</td>
<td align="left">More comprehensive</td>
<td align="left">Less comprehensive</td>
</tr>
<tr class="odd">
<td align="left">Updates</td>
<td align="left">Frequent</td>
<td align="left">Less frequent</td>
</tr>
<tr class="even">
<td align="left">Orthology databases</td>
<td align="left">Ensembl, HomoloGene, WormBase</td>
<td align="left">HomoloGene</td>
</tr>
<tr class="odd">
<td align="left">Data location</td>
<td align="left">Remote</td>
<td align="left">Local</td>
</tr>
<tr class="even">
<td align="left">Internet connection</td>
<td align="left">Required</td>
<td align="left">Not required</td>
</tr>
<tr class="odd">
<td align="left">Speed</td>
<td align="left">Slower</td>
<td align="left">Faster</td>
</tr>
</tbody>
</table>
</div>
<div id="quick-examples" class="section level1">
<h1 class="hasAnchor">
<a href="#quick-examples" class="anchor"></a>Quick examples</h1>
<div id="convert-orthologs" class="section level2">
<h2 class="hasAnchor">
<a href="#convert-orthologs" class="anchor"></a>Convert orthologs</h2>
<p><a href="https://neurogenomics.github.io/orthogene/reference/convert_orthologs.html"><code>convert_orthologs</code></a> is very flexible with what users can supply as <code>gene_df</code>, and can take a <code>data.frame</code>/<code>data.table</code>/<code>tibble</code>, (sparse) <code>matrix</code>, or <code>list</code>/<code>vector</code> containing genes.</p>
<p>Genes will be recognised in most formats (e.g. HGNC, Ensembl, UCSC) and can even be a mixture of different formats.</p>
<p>All genes will be mapped to gene symbols, unless specified otherwise with the <code>...</code> arguments (see <code><a href="reference/convert_orthologs.html">?orthogene::convert_orthologs</a></code> or <a href="https://neurogenomics.github.io/orthogene/reference/convert_orthologs.html">here</a> for details).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"exp_mouse"</span><span class="op">)</span>
<span class="va">gene_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/convert_orthologs.html">convert_orthologs</a></span><span class="op">(</span>gene_df <span class="op">=</span> <span class="va">exp_mouse</span>,
                             gene_input <span class="op">=</span> <span class="st">"rownames"</span>, 
                             gene_output <span class="op">=</span> <span class="st">"rownames"</span>, 
                             input_species <span class="op">=</span> <span class="st">"mouse"</span>,
                             method <span class="op">=</span> <span class="va">method</span><span class="op">)</span> </code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Preparing gene_df.</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="do">## sparseMatrix format detected.</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Extracting genes from rownames.</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Converting mouse ==&gt; human orthologs using: homologene</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="do">## Using stored `gprofiler_orgs`.</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Mapping species name: mouse</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Common name mapping found for mouse</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="do">## 1 organism identified from search: 10090</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a><span class="do">## Using stored `gprofiler_orgs`.</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a><span class="do">## Mapping species name: human</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a><span class="do">## Common name mapping found for human</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a><span class="do">## 1 organism identified from search: 9606</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a><span class="do">## Checking for genes without orthologs in human.</span></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a><span class="do">## Extracting genes from input_gene.</span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a><span class="do">## Extracting genes from ortholog_gene.</span></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a><span class="do">## Checking for genes without 1:1 orthologs.</span></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a><span class="do">## Dropping 46 genes that have multiple input_gene per ortholog_gene.</span></span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a><span class="do">## Dropping 56 genes that have multiple ortholog_gene per input_gene.</span></span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a><span class="do">## Filtering gene_df with gene_map</span></span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a><span class="do">## Setting ortholog_gene to rownames.</span></span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a><span class="do">## =========== REPORT SUMMARY ===========</span></span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a><span class="do">## Total genes dropped after convert_orthologs :</span></span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a><span class="do">##    2,016 / 15,259 (13%)</span></span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a><span class="do">## Total genes remaining after convert_orthologs :</span></span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a><span class="do">##    13,243 / 15,259 (87%)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">gene_df</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table">
<colgroup>
<col width="8%">
<col width="18%">
<col width="15%">
<col width="11%">
<col width="8%">
<col width="14%">
<col width="12%">
<col width="11%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">astrocytes_ependymal</th>
<th align="right">endothelial-mural</th>
<th align="right">interneurons</th>
<th align="right">microglia</th>
<th align="right">oligodendrocytes</th>
<th align="right">pyramidal CA1</th>
<th align="right">pyramidal SS</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">TSPAN12</td>
<td align="right">0.3303571</td>
<td align="right">0.5872340</td>
<td align="right">0.6413793</td>
<td align="right">0.1428571</td>
<td align="right">0.1207317</td>
<td align="right">0.2864750</td>
<td align="right">0.1453634</td>
</tr>
<tr class="even">
<td align="left">TSHZ1</td>
<td align="right">0.4285714</td>
<td align="right">0.4468085</td>
<td align="right">1.1551724</td>
<td align="right">0.4387755</td>
<td align="right">0.3621951</td>
<td align="right">0.0692226</td>
<td align="right">0.8320802</td>
</tr>
<tr class="odd">
<td align="left">ADAMTS15</td>
<td align="right">0.0089286</td>
<td align="right">0.0978723</td>
<td align="right">0.2206897</td>
<td align="right">0.0000000</td>
<td align="right">0.0231707</td>
<td align="right">0.0117146</td>
<td align="right">0.0375940</td>
</tr>
<tr class="even">
<td align="left">CLDN12</td>
<td align="right">0.2232143</td>
<td align="right">0.1148936</td>
<td align="right">0.5517241</td>
<td align="right">0.0510204</td>
<td align="right">0.2609756</td>
<td align="right">0.4376997</td>
<td align="right">0.6842105</td>
</tr>
<tr class="odd">
<td align="left">RXFP1</td>
<td align="right">0.0000000</td>
<td align="right">0.0127660</td>
<td align="right">0.2551724</td>
<td align="right">0.0000000</td>
<td align="right">0.0158537</td>
<td align="right">0.0511182</td>
<td align="right">0.0751880</td>
</tr>
<tr class="even">
<td align="left">SEMA3C</td>
<td align="right">0.1964286</td>
<td align="right">0.9957447</td>
<td align="right">8.6379310</td>
<td align="right">0.2040816</td>
<td align="right">0.1853659</td>
<td align="right">0.1608094</td>
<td align="right">0.2280702</td>
</tr>
</tbody>
</table>
</div>
<div id="map-species" class="section level2">
<h2 class="hasAnchor">
<a href="#map-species" class="anchor"></a>Map species</h2>
<p><a href="https://neurogenomics.github.io/orthogene/reference/map_species.html"><code>map_species</code></a> lets you standardise species names from a wide variety of identifiers (e.g. common name, taxonomy ID, Latin name, partial match).</p>
<p>All exposed <code>orthogene</code> functions (including <a href="https://neurogenomics.github.io/orthogene/reference/convert_orthologs.html"><code>convert_orthologs</code></a>) use <code>map_species</code> under the hood, so you don’t have to worry about getting species names exactly right.</p>
<p>You can check the full list of available species by simply running <code><a href="reference/map_species.html">map_species()</a></code> with no arguments, or checking <a href="https://biit.cs.ut.ee/gprofiler/page/organism-list">here</a>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">species</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/map_species.html">map_species</a></span><span class="op">(</span>species <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"human"</span>,<span class="fl">9544</span>,<span class="st">"mus musculus"</span>,
                                   <span class="st">"fruit fly"</span>,<span class="st">"Celegans"</span><span class="op">)</span>, 
                       output_format <span class="op">=</span> <span class="st">"scientific_name"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Using stored `gprofiler_orgs`.</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Mapping species name: human</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Common name mapping found for human</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="do">## 1 organism identified from search: Homo sapiens</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="do">## Mapping species name: 9544</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="do">## 1 organism identified from search: Macaca mulatta</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Mapping species name: mus musculus</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="do">## 1 organism identified from search: Mus musculus</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a><span class="do">## Mapping species name: fruit fly</span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a><span class="do">## Common name mapping found for fruit fly</span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a><span class="do">## 1 organism identified from search: Drosophila melanogaster</span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a><span class="do">## Mapping species name: Celegans</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a><span class="do">## 1 organism identified from search: Caenorhabditis elegans</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">species</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="do">##                     human                      9544              mus musculus </span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="do">##            "Homo sapiens"          "Macaca mulatta"            "Mus musculus" </span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="do">##                 fruit fly                  Celegans </span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="do">## "Drosophila melanogaster"  "Caenorhabditis elegans"</span></span></code></pre></div>
</div>
<div id="report-orthologs" class="section level2">
<h2 class="hasAnchor">
<a href="#report-orthologs" class="anchor"></a>Report orthologs</h2>
<p>It may be helpful to know the maximum expected number of orthologous gene mappings from one species to another.</p>
<p><a href="https://neurogenomics.github.io/orthogene/reference/report_orthologs.html"><code>ortholog_report</code></a> generates a report that tells you this information genome-wide.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orth_zeb</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/report_orthologs.html">report_orthologs</a></span><span class="op">(</span>target_species <span class="op">=</span> <span class="st">"zebrafish"</span>,
                             reference_species <span class="op">=</span> <span class="st">"human"</span>,
                             method_all_genes <span class="op">=</span> <span class="va">method</span>,
                             method_convert_orthologs <span class="op">=</span> <span class="va">method</span><span class="op">)</span> </code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Retrieving all genes using: homologene.</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Using stored `gprofiler_orgs`.</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Mapping species name: zebrafish</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Common name mapping found for zebrafish</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="do">## 1 organism identified from search: 7955</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Gene table with 20897 rows retrieved.</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Extracting genes from Gene.Symbol.</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Returning all 20,897 genes from zebrafish.</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a><span class="do">## Retrieving all genes using: homologene.</span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a><span class="do">## Using stored `gprofiler_orgs`.</span></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a><span class="do">## Mapping species name: human</span></span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a><span class="do">## Common name mapping found for human</span></span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a><span class="do">## 1 organism identified from search: 9606</span></span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a><span class="do">## Gene table with 19129 rows retrieved.</span></span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a><span class="do">## Extracting genes from Gene.Symbol.</span></span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a><span class="do">## Returning all 19,129 genes from human.</span></span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a><span class="do">## Preparing gene_df.</span></span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a><span class="do">## data.frame format detected.</span></span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a><span class="do">## Extracting genes from Gene.Symbol.</span></span>
<span id="cb11-38"><a href="#cb11-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-39"><a href="#cb11-39" aria-hidden="true" tabindex="-1"></a><span class="do">## Converting zebrafish ==&gt; human orthologs using: homologene</span></span>
<span id="cb11-40"><a href="#cb11-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-41"><a href="#cb11-41" aria-hidden="true" tabindex="-1"></a><span class="do">## Using stored `gprofiler_orgs`.</span></span>
<span id="cb11-42"><a href="#cb11-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-43"><a href="#cb11-43" aria-hidden="true" tabindex="-1"></a><span class="do">## Mapping species name: zebrafish</span></span>
<span id="cb11-44"><a href="#cb11-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-45"><a href="#cb11-45" aria-hidden="true" tabindex="-1"></a><span class="do">## Common name mapping found for zebrafish</span></span>
<span id="cb11-46"><a href="#cb11-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-47"><a href="#cb11-47" aria-hidden="true" tabindex="-1"></a><span class="do">## 1 organism identified from search: 7955</span></span>
<span id="cb11-48"><a href="#cb11-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-49"><a href="#cb11-49" aria-hidden="true" tabindex="-1"></a><span class="do">## Using stored `gprofiler_orgs`.</span></span>
<span id="cb11-50"><a href="#cb11-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-51"><a href="#cb11-51" aria-hidden="true" tabindex="-1"></a><span class="do">## Mapping species name: human</span></span>
<span id="cb11-52"><a href="#cb11-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-53"><a href="#cb11-53" aria-hidden="true" tabindex="-1"></a><span class="do">## Common name mapping found for human</span></span>
<span id="cb11-54"><a href="#cb11-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-55"><a href="#cb11-55" aria-hidden="true" tabindex="-1"></a><span class="do">## 1 organism identified from search: 9606</span></span>
<span id="cb11-56"><a href="#cb11-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-57"><a href="#cb11-57" aria-hidden="true" tabindex="-1"></a><span class="do">## Checking for genes without orthologs in human.</span></span>
<span id="cb11-58"><a href="#cb11-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-59"><a href="#cb11-59" aria-hidden="true" tabindex="-1"></a><span class="do">## Extracting genes from input_gene.</span></span>
<span id="cb11-60"><a href="#cb11-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-61"><a href="#cb11-61" aria-hidden="true" tabindex="-1"></a><span class="do">## Extracting genes from ortholog_gene.</span></span>
<span id="cb11-62"><a href="#cb11-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-63"><a href="#cb11-63" aria-hidden="true" tabindex="-1"></a><span class="do">## Checking for genes without 1:1 orthologs.</span></span>
<span id="cb11-64"><a href="#cb11-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-65"><a href="#cb11-65" aria-hidden="true" tabindex="-1"></a><span class="do">## Dropping 47 genes that have multiple input_gene per ortholog_gene.</span></span>
<span id="cb11-66"><a href="#cb11-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-67"><a href="#cb11-67" aria-hidden="true" tabindex="-1"></a><span class="do">## Dropping 2,708 genes that have multiple ortholog_gene per input_gene.</span></span>
<span id="cb11-68"><a href="#cb11-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-69"><a href="#cb11-69" aria-hidden="true" tabindex="-1"></a><span class="do">## Filtering gene_df with gene_map</span></span>
<span id="cb11-70"><a href="#cb11-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-71"><a href="#cb11-71" aria-hidden="true" tabindex="-1"></a><span class="do">## Adding input_gene col to gene_df.</span></span>
<span id="cb11-72"><a href="#cb11-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-73"><a href="#cb11-73" aria-hidden="true" tabindex="-1"></a><span class="do">## Adding ortholog_gene col to gene_df.</span></span>
<span id="cb11-74"><a href="#cb11-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-75"><a href="#cb11-75" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb11-76"><a href="#cb11-76" aria-hidden="true" tabindex="-1"></a><span class="do">## =========== REPORT SUMMARY ===========</span></span>
<span id="cb11-77"><a href="#cb11-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-78"><a href="#cb11-78" aria-hidden="true" tabindex="-1"></a><span class="do">## Total genes dropped after convert_orthologs :</span></span>
<span id="cb11-79"><a href="#cb11-79" aria-hidden="true" tabindex="-1"></a><span class="do">##    10,338 / 20,895 (49%)</span></span>
<span id="cb11-80"><a href="#cb11-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-81"><a href="#cb11-81" aria-hidden="true" tabindex="-1"></a><span class="do">## Total genes remaining after convert_orthologs :</span></span>
<span id="cb11-82"><a href="#cb11-82" aria-hidden="true" tabindex="-1"></a><span class="do">##    10,557 / 20,895 (51%)</span></span>
<span id="cb11-83"><a href="#cb11-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-84"><a href="#cb11-84" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb11-85"><a href="#cb11-85" aria-hidden="true" tabindex="-1"></a><span class="do">## =========== REPORT SUMMARY ===========</span></span>
<span id="cb11-86"><a href="#cb11-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-87"><a href="#cb11-87" aria-hidden="true" tabindex="-1"></a><span class="do">## 10,556 / 20,895 (50.52%) target_species genes remain after ortholog conversion.</span></span>
<span id="cb11-88"><a href="#cb11-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-89"><a href="#cb11-89" aria-hidden="true" tabindex="-1"></a><span class="do">## 10,556 / 19,129 (55.18%) reference_species genes remain after ortholog conversion.</span></span></code></pre></div>
</div>
<div id="map-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#map-genes" class="anchor"></a>Map genes</h2>
<p><a href="https://neurogenomics.github.io/orthogene/reference/map_genes.html"><code>map_genes</code></a> finds matching <em>within-species</em> synonyms across a wide variety of gene naming conventions (e.g. HGNC symbols, ENSEMBL IDs, UCSC) and returns a table with standardised gene symbols (or whatever output format you prefer).</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mapped_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/map_genes.html">map_genes</a></span><span class="op">(</span>genes<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">exp_mouse</span><span class="op">)</span>,
                          species<span class="op">=</span><span class="st">"mouse"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Using stored `gprofiler_orgs`.</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Mapping species name: mouse</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Common name mapping found for mouse</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="do">## 1 organism identified from search: mmusculus</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="do">## 15,259 / 15,397 (100.9%) genes mapped.</span></span></code></pre></div>
</div>
<div id="get-all-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#get-all-genes" class="anchor"></a>Get all genes</h2>
<p>You can also quickly get all known genes from the genome of a given species with <a href="https://neurogenomics.github.io/orthogene/reference/all_genes.html"><code>all_genes</code></a>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genome_mouse</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/all_genes.html">all_genes</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"mouse"</span>, 
                          method <span class="op">=</span> <span class="va">method</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Retrieving all genes using: homologene.</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Using stored `gprofiler_orgs`.</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Mapping species name: mouse</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Common name mapping found for mouse</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="do">## 1 organism identified from search: 10090</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Gene table with 21207 rows retrieved.</span></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Extracting genes from Gene.Symbol.</span></span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Returning all 21,207 genes from mouse.</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">genome_mouse</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">HID</th>
<th align="right">Gene.ID</th>
<th align="left">Gene.Symbol</th>
<th align="right">Taxonomy</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">6</td>
<td align="right">3</td>
<td align="right">11364</td>
<td align="left">Acadm</td>
<td align="right">10090</td>
</tr>
<tr class="even">
<td align="left">18</td>
<td align="right">5</td>
<td align="right">11370</td>
<td align="left">Acadvl</td>
<td align="right">10090</td>
</tr>
<tr class="odd">
<td align="left">29</td>
<td align="right">6</td>
<td align="right">110446</td>
<td align="left">Acat1</td>
<td align="right">10090</td>
</tr>
<tr class="even">
<td align="left">52</td>
<td align="right">7</td>
<td align="right">11477</td>
<td align="left">Acvr1</td>
<td align="right">10090</td>
</tr>
<tr class="odd">
<td align="left">64</td>
<td align="right">9</td>
<td align="right">20391</td>
<td align="left">Sgca</td>
<td align="right">10090</td>
</tr>
<tr class="even">
<td align="left">71</td>
<td align="right">12</td>
<td align="right">11564</td>
<td align="left">Adsl</td>
<td align="right">10090</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<details><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="do">## R version 4.1.0 (2021-05-18)</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Running under: Ubuntu 20.04.2 LTS</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Matrix products: default</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="do">## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="do">## locale:</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a><span class="do">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a><span class="do">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a><span class="do">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             </span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a><span class="do">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a><span class="do">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a><span class="do">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a><span class="do">## attached base packages:</span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a><span class="do">## other attached packages:</span></span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] orthogene_0.1.0</span></span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a><span class="do">## loaded via a namespace (and not attached):</span></span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a><span class="do">##  [1] Rcpp_1.0.7                lattice_0.20-44          </span></span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a><span class="do">##  [3] tidyr_1.1.3               assertthat_0.2.1         </span></span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a><span class="do">##  [5] digest_0.6.27             utf8_1.2.2               </span></span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a><span class="do">##  [7] R6_2.5.0                  cellranger_1.1.0         </span></span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a><span class="do">##  [9] backports_1.2.1           evaluate_0.14            </span></span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a><span class="do">## [11] httr_1.4.2                ggplot2_3.3.5            </span></span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a><span class="do">## [13] highr_0.9                 pillar_1.6.1             </span></span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a><span class="do">## [15] rlang_0.4.11              lazyeval_0.2.2           </span></span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a><span class="do">## [17] curl_4.3.2                readxl_1.3.1             </span></span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a><span class="do">## [19] data.table_1.14.0         car_3.0-11               </span></span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a><span class="do">## [21] Matrix_1.3-4              rmarkdown_2.9            </span></span>
<span id="cb18-34"><a href="#cb18-34" aria-hidden="true" tabindex="-1"></a><span class="do">## [23] stringr_1.4.0             foreign_0.8-81           </span></span>
<span id="cb18-35"><a href="#cb18-35" aria-hidden="true" tabindex="-1"></a><span class="do">## [25] htmlwidgets_1.5.3         RCurl_1.98-1.3           </span></span>
<span id="cb18-36"><a href="#cb18-36" aria-hidden="true" tabindex="-1"></a><span class="do">## [27] munsell_0.5.0             broom_0.7.8              </span></span>
<span id="cb18-37"><a href="#cb18-37" aria-hidden="true" tabindex="-1"></a><span class="do">## [29] compiler_4.1.0            gprofiler2_0.2.0         </span></span>
<span id="cb18-38"><a href="#cb18-38" aria-hidden="true" tabindex="-1"></a><span class="do">## [31] xfun_0.24                 pkgconfig_2.0.3          </span></span>
<span id="cb18-39"><a href="#cb18-39" aria-hidden="true" tabindex="-1"></a><span class="do">## [33] htmltools_0.5.1.1         tidyselect_1.1.1         </span></span>
<span id="cb18-40"><a href="#cb18-40" aria-hidden="true" tabindex="-1"></a><span class="do">## [35] tibble_3.1.3              rio_0.5.27               </span></span>
<span id="cb18-41"><a href="#cb18-41" aria-hidden="true" tabindex="-1"></a><span class="do">## [37] fansi_0.5.0               viridisLite_0.4.0        </span></span>
<span id="cb18-42"><a href="#cb18-42" aria-hidden="true" tabindex="-1"></a><span class="do">## [39] crayon_1.4.1              dplyr_1.0.7              </span></span>
<span id="cb18-43"><a href="#cb18-43" aria-hidden="true" tabindex="-1"></a><span class="do">## [41] ggpubr_0.4.0              bitops_1.0-7             </span></span>
<span id="cb18-44"><a href="#cb18-44" aria-hidden="true" tabindex="-1"></a><span class="do">## [43] grid_4.1.0                jsonlite_1.7.2           </span></span>
<span id="cb18-45"><a href="#cb18-45" aria-hidden="true" tabindex="-1"></a><span class="do">## [45] gtable_0.3.0              lifecycle_1.0.0          </span></span>
<span id="cb18-46"><a href="#cb18-46" aria-hidden="true" tabindex="-1"></a><span class="do">## [47] DBI_1.1.1                 magrittr_2.0.1           </span></span>
<span id="cb18-47"><a href="#cb18-47" aria-hidden="true" tabindex="-1"></a><span class="do">## [49] scales_1.1.1              zip_2.2.0                </span></span>
<span id="cb18-48"><a href="#cb18-48" aria-hidden="true" tabindex="-1"></a><span class="do">## [51] stringi_1.7.3             carData_3.0-4            </span></span>
<span id="cb18-49"><a href="#cb18-49" aria-hidden="true" tabindex="-1"></a><span class="do">## [53] ggsignif_0.6.2            ellipsis_0.3.2           </span></span>
<span id="cb18-50"><a href="#cb18-50" aria-hidden="true" tabindex="-1"></a><span class="do">## [55] generics_0.1.0            vctrs_0.3.8              </span></span>
<span id="cb18-51"><a href="#cb18-51" aria-hidden="true" tabindex="-1"></a><span class="do">## [57] openxlsx_4.2.4            tools_4.1.0              </span></span>
<span id="cb18-52"><a href="#cb18-52" aria-hidden="true" tabindex="-1"></a><span class="do">## [59] forcats_0.5.1             homologene_1.4.68.19.3.27</span></span>
<span id="cb18-53"><a href="#cb18-53" aria-hidden="true" tabindex="-1"></a><span class="do">## [61] glue_1.4.2                purrr_0.3.4              </span></span>
<span id="cb18-54"><a href="#cb18-54" aria-hidden="true" tabindex="-1"></a><span class="do">## [63] hms_1.1.0                 abind_1.4-5              </span></span>
<span id="cb18-55"><a href="#cb18-55" aria-hidden="true" tabindex="-1"></a><span class="do">## [65] parallel_4.1.0            yaml_2.2.1               </span></span>
<span id="cb18-56"><a href="#cb18-56" aria-hidden="true" tabindex="-1"></a><span class="do">## [67] colorspace_2.0-2          rstatix_0.7.0            </span></span>
<span id="cb18-57"><a href="#cb18-57" aria-hidden="true" tabindex="-1"></a><span class="do">## [69] plotly_4.9.4.9000         knitr_1.33               </span></span>
<span id="cb18-58"><a href="#cb18-58" aria-hidden="true" tabindex="-1"></a><span class="do">## [71] haven_2.4.1               patchwork_1.1.1</span></span></code></pre></div>
</details>
</div>
<div id="related-projects" class="section level1">
<h1 class="hasAnchor">
<a href="#related-projects" class="anchor"></a>Related projects</h1>
<div id="tools" class="section level2">
<h2 class="hasAnchor">
<a href="#tools" class="anchor"></a>Tools</h2>
<ul>
<li><p><a href="https://github.com/oganm/homologene"><code>homologene</code></a>: <code>orthogene</code> uses this package. Provides API access to NCBI <a href="https://www.ncbi.nlm.nih.gov/homologene">HomoloGene</a> database.</p></li>
<li><p><a href="https://cran.r-project.org/web/packages/gprofiler2/vignettes/gprofiler2.html"><code>gprofiler2</code></a>: <code>orthogene</code> uses this package. <code><a href="https://rdrr.io/pkg/gprofiler2/man/gorth.html">gprofiler2::gorth()</a></code> pulls from <a href="https://biit.cs.ut.ee/gprofiler/page/organism-list">many orthology mapping databases</a>.</p></li>
<li><p><a href="https://www.bioconductor.org/packages/release/bioc/html/annotationTools.html"><code>annotationTools</code></a>: For interspecies microarray data.</p></li>
<li><p><a href="https://www.leibniz-hki.de/en/orthology-r-package.html"><code>orthology</code></a>: R package for ortholog mapping (deprecated?).</p></li>
<li><p><a href="https://rdrr.io/github/elsayed-lab/hpgltools/man/load_biomart_orthologs.html"><code>hpgltools::load_biomart_orthologs()</code></a>: Helper function to get orthologs from biomart.</p></li>
<li><p><a href="https://github.com/ridgelab/JustOrthologs/"><code>JustOrthologs</code></a>: Ortholog inference from multi-species genomic sequences.</p></li>
<li><p><a href="https://github.com/drostlab/orthologr"><code>orthologr</code></a>: Ortholog inference from multi-species genomic sequences.</p></li>
<li><p><a href="https://github.com/davidemms/OrthoFinder"><code>OrthoFinder</code></a>: Gene duplication event inference from multi-species genomics.</p></li>
</ul>
</div>
<div id="databases" class="section level2">
<h2 class="hasAnchor">
<a href="#databases" class="anchor"></a>Databases</h2>
<ul>
<li><p><a href="https://www.ncbi.nlm.nih.gov/homologene">HomoloGene</a>: NCBI database that the R package <a href="https://github.com/oganm/homologene">homologene</a> pulls from.</p></li>
<li><p><a href="https://biit.cs.ut.ee/gprofiler">gProfiler</a>: Web server for functional enrichment analysis and conversions of gene lists.</p></li>
<li><p><a href="http://orthologene.org/resources.html">OrtholoGene</a>: Compiled list of gene orthology resources.</p></li>
</ul>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/neurogenomics/orthogene/">https://​github.com/​neurogenomics/​orthogene/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/neurogenomics/orthogene/issues%20">https://​github.com/​neurogenomics/​orthogene/​issues </a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a> </li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Brian Schilder <br><small class="roles"> Maintainer </small> <a href="https://orcid.org/0000-0001-5949-2191" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://codecov.io/gh/neurogenomics/orthogene"><img src="https://codecov.io/gh/neurogenomics/orthogene/branch/main/graph/badge.svg"></a></li>
<li><a href="https://github.com/neurogenomics/MungeSumstats/actions"><img src="https://github.com/neurogenomics/MungeSumstats/workflows/R-full/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="https://img.shields.io/badge/lifecycle-stable-green.svg"></a></li>
<li><a href="https://github.com/neurogenomics/orthogene/commits/main"><img src="https://img.shields.io/github/last-commit/neurogenomics/orthogene.svg"></a></li>
<li><a href="https://cran.r-project.org/web/licenses/Artistic-2.0"><img src="https://img.shields.io/badge/license-Artistic--2.0-blue.svg" alt="License: Artistic-2.0"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
